schema/schema.py

```diff
@@ -0,0 +1,86 @@
+"""Handles JSON schema generation logic"""
+
+
+import importlib
+
+import click
+
+from samcli.cli.command import _SAM_CLI_COMMAND_PACKAGES
+from samcli.lib.config.samconfig import SamConfig
+
+
+def format_param(param: click.core.Option):
+    """Format a click Option parameter to a dictionary object."""
+    formatted_param = {"name": param.name, "help": param.help}
+
+    if param.type.name in ["text", "path", "choice"]:
+        formatted_param["type"] = "string"
+    else:
+        formatted_param["type"] = param.type.name
+
+    if param.default:
+        formatted_param["default"] = param.default
+
+    if param.type.name == "choice":
+        formatted_param["choices"] = param.type.choices
+
+    return formatted_param
+
+
+def retrieve_command_structure(package_name: str) -> dict:
+    """Given a SAM CLI package name, retrieve its structure.
+
+    Returns
+    -------
+    dict
+        A dictionary that maps the name of the command to its relevant click options.
+    """
+    module = importlib.import_module(package_name)
+    command = {}
+
+    def get_params_from_command(cli, main_command_name: str = "") -> dict:
```

Comment:
question: any particular reasoning as to why we have this as an inner method? No particular issue with this aside from making it easier to test if this was located outside the retrieve method.

Reply:
My line of reasoning was that it was probably only going to be used within the context of this method as a helper function, but we could bring it outside if that's preferred

Reply:
Let's move it outside, that way when it comes time to test things, we can isolate this method to make it easier to test